### PR TITLE
ExportNotesInMargin allows to export comments in margin

### PR DESCRIPTION
**Pull request by maniack**
Hello!
I needed to display comments in a margin... So I decided to add this option to your app.
I think this is a useful one :)

https://help.libreoffice.org/latest/en-US/text/shared/guide/pdf_params.html

sw_text_with_comment

### CODE REVIEW

File: pkg/modules/libreoffice/api/api.go [hunk 1]
```diff
@@ -69,6 +69,10 @@ type Options struct {
 	// Optional
 	ReduceImageResolution bool
 
+	// ExportNotesInMargin allows to export comments in margin
+	// Optional
+	ExportNotesInMargin bool
```
Comment (gulien) [inline pkg/modules/libreoffice/api/api.go:74]:
```suggestion
	// ExportNotesInMargin allows to export comments in margin.
	// Optional
	ExportNotesInMargin bool
```

Comment (gulien) [reply pkg/modules/libreoffice/api/api.go:74]: Could you also move this before `LosslessImageCompression`?

File: pkg/modules/libreoffice/api/libreoffice.go [hunk 2]
```diff
@@ -289,6 +289,14 @@ func (p *libreOfficeProcess) pdf(ctx context.Context, logger *zap.Logger, inputP
 		args = append(args, "--export", "ReduceImageResolution=false")
 	}
 
+	if options.ExportNotesInMargin {
+		args = append(
+			args,
+			"--export", "ExportNotes=true",
+			"--export", "ExportNotesInMargin=true",
```
Comment (gulien) [inline pkg/modules/libreoffice/api/libreoffice.go:296]: What is `ExportNotes` here 🤔?

File: pkg/modules/libreoffice/api/libreoffice_test.go [hunk 3]
```diff
@@ -509,6 +509,35 @@ func TestLibreOfficeProcess_pdf(t *testing.T) {
 			start:        true,
 			expectError:  false,
 		},
+		{
+			scenario: "success ExportNotesInMargin",
```
Comment (gulien) [inline pkg/modules/libreoffice/api/libreoffice_test.go:513]: Could you also move this before `LosslessImageCompression`?

File: pkg/modules/libreoffice/routes.go [hunk 4]
```diff
@@ -32,6 +32,7 @@ func convertRoute(libreOffice libreofficeapi.Uno, engine gotenberg.PdfEngine) ap
 				singlePageSheets         bool
 				losslessImageCompression bool
 				reduceImageResolution    bool
+				exportNotesInMargin      bool
```
Comment (gulien) [inline pkg/modules/libreoffice/routes.go:35]: Could you also move this before `LosslessImageCompression`?

File: pkg/modules/libreoffice/routes.go [hunk 5]
```diff
@@ -47,6 +48,7 @@ func convertRoute(libreOffice libreofficeapi.Uno, engine gotenberg.PdfEngine) ap
 				Bool("singlePageSheets", &singlePageSheets, false).
 				Bool("losslessImageCompression", &losslessImageCompression, false).
 				Bool("reduceImageResolution", &reduceImageResolution, true).
+				Bool("exportNotesInMargin", &exportNotesInMargin, false).
```
Comment (gulien) [inline pkg/modules/libreoffice/routes.go:51]: Could you also move this before `LosslessImageCompression`?

File: pkg/modules/libreoffice/routes.go [hunk 6]
```diff
@@ -81,6 +83,7 @@ func convertRoute(libreOffice libreofficeapi.Uno, engine gotenberg.PdfEngine) ap
 					SinglePageSheets:         singlePageSheets,
 					LosslessImageCompression: losslessImageCompression,
 					ReduceImageResolution:    reduceImageResolution,
+					ExportNotesInMargin:      exportNotesInMargin,
```
Comment (gulien) [inline pkg/modules/libreoffice/routes.go:86]: Could you also move this before `LosslessImageCompression`?

**Review by gulien — CHANGES_REQUESTED** [review]
Thank you very much for this PR!

File: pkg/modules/libreoffice/api/libreoffice.go (hunk 2 above)
Comment (maniack) [reply pkg/modules/libreoffice/api/libreoffice.go:296]: As it said in LibreOffice [documentation](https://help.libreoffice.org/latest/en-US/text/shared/guide/pdf_params.html): 
```ExportNotes - Specifies if notes are exported to PDF, boolean, default false```
So to be sure that we're exporting notes at all  to show them in margin I decided to add this overhead 😇
Am I wrong?

Comment (gulien) [reply pkg/modules/libreoffice/api/libreoffice.go:296]: I’m wondering is there are another kinds of notes than « margin » ones ^^ like would it make sense to have an `exportNotes` form fields?

Comment (maniack) [reply pkg/modules/libreoffice/api/libreoffice.go]: I've done some research and testing, there are at least two options for displaying notes/comments:
`ExportNotes` - show notes "inline", like a pinned note at the end of a modified sentence and
`ExportNotesInMargin` - show notes in the margins next to the modified line.

Sorry for the confusion :( I made the first changes hastily and didn't go deep into details.

File: pkg/modules/libreoffice/api/libreoffice_test.go (hunk 3 above)
Comment (maniack) [reply pkg/modules/libreoffice/api/libreoffice_test.go]: Done

File: pkg/modules/libreoffice/routes.go (hunk 4 above)
Comment (maniack) [reply pkg/modules/libreoffice/routes.go]: Done

File: pkg/modules/libreoffice/routes.go (hunk 5 above)
Comment (maniack) [reply pkg/modules/libreoffice/routes.go]: Done

File: pkg/modules/libreoffice/routes.go (hunk 6 above)
Comment (maniack) [reply pkg/modules/libreoffice/routes.go]: Done

File: pkg/modules/libreoffice/api/api.go (hunk 1 above)
Comment (maniack) [reply pkg/modules/libreoffice/api/api.go]: Done

**Review by gulien — APPROVED** [review]

**Comment by gulien** [comment]
@maniack `gotenberg/gotenberg:edge` should be available within ~1 hour 👍 thanks again!

**Pull request merged**